app/controllers/publishers/ats_api/v1/create_vacancy_service.rb

```diff
@@ -0,0 +1,77 @@
+module Publishers
+  module AtsApi
+    module V1
+      class CreateVacancyService
+        InvalidOrganisationError = Class.new(StandardError)
+
+        class << self
+          def call(params)
+            vacancy = Vacancy.new(sanitised_params(params))
+
+            if (conflict = conflict_vacancy(vacancy))
+              return conflict_response(conflict)
+            end
+
+            if vacancy.save
+              success_response(vacancy)
+            else
+              validation_error_response(vacancy)
+            end
+          end
+
+          private
+
+          def sanitised_params(params)
+            organisations = fetch_organisations(params[:schools])
+            raise InvalidOrganisationError, "No valid organisations found" if organisations.blank?
+
+            params[:publish_on] ||= Time.zone.today.to_s
+            params[:working_patterns] ||= []
```

Comment:
blank working patterns is an invalid vacancy - so I'm not sure what the intent of this line is?